### PR TITLE
dates and timestamps should be human-readable [AJ-620]

### PR DESCRIPTION
**Pull request by davidangb**
* Datetime attributes now serialize to e.g. `2022-04-25T12:00:01`
* Date attributes now serialize to e.g. `2022-04-25`
* The `RecordAttributes` underlying `Map` now contains `LocalDate` and `LocalDateTime` values

### CODE REVIEW

File: service/src/main/java/org/databiosphere/workspacedataservice/dao/RecordDao.java [hunk 1]
```diff
@@ -417,8 +417,16 @@ private RecordAttributes getAttributes(ResultSet rs) {
...
+
+						if (object instanceof java.sql.Date date) {
+							object = date.toString();
+						} else if (object instanceof java.sql.Timestamp ts) {
+							object = ts.toString();
+						} else if (object instanceof PGobject pGobject) {
+							object = pGobject.getValue();
+						}
+
+						attributes.putAttribute(columnName, object);
```
Comment (sagehen03) [inline service/src/main/java/org/databiosphere/workspacedataservice/dao/RecordDao.java:429]: This approach is one step closer to Map<String, String> to keep the attributes in memory.  The current functionality doesn't require anything more, but we might need it at some point.  We can use something like `Object object = rs.getObject(columnName, schema.get(columnName).getJavaType());` to get the properly typed java objects into the attribute map, but it does require the schema info in the RowMapper.

Comment (davidangb) [reply service/src/main/java/org/databiosphere/workspacedataservice/dao/RecordDao.java:429]: An approach that would keep the real java types in the attribute map:
```java
						if (object instanceof java.sql.Date date) {
							object = date.toLocalDate();
						} else if (object instanceof java.sql.Timestamp ts) {
							object = ts.toLocalDateTime();
						} else if (object instanceof PGobject pGobject) {
							object = pGobject.getValue();
						}
```

If we do that, then we just need to configure the `ObjectMapper` to serialize `LocalDate` and `LocalDateTime`

File: service/src/main/java/org/databiosphere/workspacedataservice/WorkspaceDataServiceApplication.java [hunk 2]
```diff
@@ -14,7 +15,8 @@ public class WorkspaceDataServiceApplication {
 
 	@Bean
 	public ObjectMapper objectMapper() {
-		return JsonMapper.builder().enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS).build();
+		return JsonMapper.builder().enable(MapperFeature.ACCEPT_CASE_INSENSITIVE_ENUMS)
+				.configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false).findAndAddModules().build();
```
Comment (davidangb) [inline service/src/main/java/org/databiosphere/workspacedataservice/WorkspaceDataServiceApplication.java:19]: the `.findAndAddModules()` could be reduced to just `mapper.registerModule(new JavaTimeModule());` but I believe the former is the default Spring behavior, so I'm choosing that approach.

File: service/src/test/java/org/databiosphere/workspacedataservice/controller/FullStackRecordControllerTest.java [hunk 3]
```diff
@@ -126,10 +127,12 @@ void testSortedQuerySuccess() throws Exception {
...
-		assertThat(Instant.ofEpochMilli((Long)body.records().get(0).recordAttributes().getAttributeValue("attr-dt")).atZone(ZoneId.systemDefault()).toLocalDate()).as("Record with attr-dt 2022-04-25 should be first record in descending order")
-				.isEqualTo("2022-04-25");
-		assertThat(Instant.ofEpochMilli((Long)body.records().get(4).recordAttributes().getAttributeValue("attr-dt")).atZone(ZoneId.systemDefault()).toLocalDate())
-				.isEqualTo("2002-03-23");
+
+		assertThat(body.records().get(0).recordAttributes().getAttributeValue("attr-dt"))
+				.as("Record with attr-dt 2022-04-25 should be first record in descending order")
+				.isEqualTo("2022-04-25T12:00:01");
+		assertThat(body.records().get(4).recordAttributes().getAttributeValue("attr-dt"))
+				.isEqualTo("2002-03-23T12:00:01");
```
Comment (davidangb) [inline service/src/test/java/org/databiosphere/workspacedataservice/controller/FullStackRecordControllerTest.java:135]: this test expected datetimes to be returned as epoch millis; they are now returned as human-readable strings.

**Review by sagehen03 — APPROVED** [review]

**Pull request merged**